Cannot download kernel headers for correct kernel version on Fedora CoreOS

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks! -->

**Describe the bug**

<!-- A clear and concise description of what the bug is. -->
I've managed to run the Falco agent in a container (Kubernetes) on Fedora CoreOS 40 with `ebfp` driver type, but after an OS upgrade from version `40.20240701.3.0` to `40.20240709.3.1` (which also updated the kernel), I started seeing the following error in the container that's trying to install the driver.

```
* Setting up /usr/src links from host
2024-08-10 08:24:50 INFO  Running falcoctl driver config
                      ├ name: falco
                      ├ version: 7.2.0+driver
                      ├ type: ebpf
                      ├ host-root: /host
                      └ repos: https://download.falco.org/driver
2024-08-10 08:24:50 INFO  Committing driver config to k8s configmap
                      └ namespace: default
2024-08-10 08:24:50 INFO  Storing falcoctl driver config
2024-08-10 08:24:50 INFO  Running falcoctl driver install
                      ├ driver version: 7.2.0+driver
                      ├ driver type: ebpf
                      ├ driver name: falco
                      ├ compile: true
                      ├ download: true
                      ├ target: fedora
                      ├ arch: x86_64
                      ├ kernel release: 6.9.7-200.fc40.x86_64
                      └ kernel version: #1 SMP PREEMPT_DYNAMIC Thu Jun 27 18:11:45 UTC 2024
2024-08-10 08:24:50 INFO  Removing eBPF probe symlink
                      └ path: /root/.falco/falco-bpf.o
2024-08-10 08:24:50 INFO  Trying to download a driver.
                      └ url: https://download.falco.org/driver/7.2.0%2Bdriver/x86_64/falco_fedora_6.9.7-200.fc40.x86_64_1.o
2024-08-10 08:24:50 WARN  Non-200 response from url. code: 404
2024-08-10 08:24:50 WARN  unable to find a prebuilt driver
2024-08-10 08:24:50 INFO  Trying to compile the requested driver
2024-08-10 08:24:50 INFO  Trying automatic kernel headers download.
2024-08-10 08:24:51 INFO  Trying to build eBPF probe.
+ cd /usr/src/falco-7.2.0+driver
+ echo '* Building eBPF probe'
* Building eBPF probe
+ '[' '!' -d /sys/kernel/debug/tracing ']'
+ echo '* Mounting debugfs'
* Mounting debugfs
+ mount -t debugfs nodev /sys/kernel/debug
+ cd bpf
+ make
make -C /lib/modules/6.9.7-200.fc40.x86_64/build M=$PWD
make[1]: *** /lib/modules/6.9.7-200.fc40.x86_64/build: No such file or directory.  Stop.
make: *** [Makefile:23: all] Error 2
2024-08-10 08:24:51 ERROR failed: failed to build all requested drivers
```

After a bit of debugging I found out that while trying to download the kernel headers, it makes a request to
https://mirrors.kernel.org/fedora/updates/40/Everything/x86_64/Packages/k/kernel-devel-6.9.7-200.fc40.x86_64.rpm
which does not exist. I think this is because the mainline Fedora distro (i.e. not CoreOS) has already moved to a newer kernel (6.10.x at the time of writing).

For reference, the URLs are set here https://github.com/falcosecurity/driverkit/blob/9ee5859bb67b56a8b114ce0ac2a0da7cd059ea89/pkg/driverbuilder/builder/fedora.go#L66-L88

If I take a look at the configured package repositories on the CoreOS host I see an interesting one:
```
❯ cat /etc/yum.repos.d/fedora-updates-archive.repo
# This is a repo that contains all the old update packages from the
# Fedora updates yum repository (i.e. the packages that have made it
# to "stable"). This repo is needed for OSTree based systems where users
# may be trying to layer packages on top of a base layer that doesn't
# have the latest stable content. Since base layer content is locked
# the package layering operation will fail unless there are older versions
# of packages available.
#
# This repo is given a high cost in order to prefer the normal Fedora
# yum repositories, which means only packages that can't be found
# elsewhere will be downloaded from here.
[updates-archive]
name=Fedora $releasever - $basearch - Updates Archive
baseurl=https://fedoraproject-updates-archive.fedoraproject.org/fedora/$releasever/$basearch/
enabled=1
metadata_expire=6h
repo_gpgcheck=0
type=rpm
gpgcheck=1
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-fedora-$releasever-$basearch
skip_if_unavailable=True
cost=10000 # default is 1000
```

Trying the `baseurl` https://fedoraproject-updates-archive.fedoraproject.org/fedora/40/x86_64/kernel-devel-6.9.7-200.fc40.x86_64.rpm retrieves the correct headers.

**How to reproduce it**

<!-- Minimal and precise steps to reproduce the bug. -->

Run latest stable Fedora CoreOS 40 (see bug description for precise version).

Start container:
`sudo podman run --rm -it --entrypoint bash --privileged --security-opt label=disable --mount type=bind,source=/proc,target=/host/proc,readonly --mount type=bind,source=/boot,target=/host/boot,readonly --mount type=bind,source=/lib/modules,target=/host/lib/modules --mount type=bind,source=/usr,target=/host/usr,readonly --mount type=bind,source=/etc,target=/host/etc,readonly docker.io/falcosecurity/falco-driver-loader:latest@sha256:5d42f7abee1d1bcf82f0cfb322cafeda7ae083b5fbe6cf69cfbabca76e5fc874`

Once in container:
```
falcoctl driver config --type ebpf
falcoctl driver install --compile=true --http-insecure=false --log-level=trace
```

**Expected behaviour**

Correct kernel headers are downloaded and driver is installed.

If kernel headers could not be found in any way, maybe show a clearer description of the error or a warning about not being able to download the headers.

**Screenshots**

N/A

**Environment**

<!-- Please complete the following info. -->

- Falco version:
<!-- Use "falco --version". -->
```2024-08-10T08:48:58+0000: Falco version: 0.38.1 (x86_64)
2024-08-10T08:48:58+0000: Falco initialized with configuration files:
2024-08-10T08:48:58+0000:    /etc/falco/falco.yaml
2024-08-10T08:48:58+0000: System info: Linux version 6.9.7-200.fc40.x86_64 (mockbuild@8d858239ee7c403c892e8a84096b3ce3) (gcc (GCC) 14.1.1 20240620 (Red Hat 14.1.1-6), GNU ld version 2.41-37.fc40) #1 SMP PREEMPT_DYNAMIC Thu Jun 27 18:11:45 UTC 2024
Falco version: 0.38.1
Libs version:  0.17.2
Plugin API:    3.6.0
Engine:        0.40.0
Driver:
  API version:    8.0.0
  Schema version: 2.0.0
  Default driver: 7.2.0+driver
```
- System info:
<!-- Falco has a built-in support command you can use  "falco --support | jq .system_info" -->
```json
{
  "machine": "x86_64",
  "nodename": "254ea178fc6a",
  "release": "6.9.7-200.fc40.x86_64",
  "sysname": "Linux",
  "version": "#1 SMP PREEMPT_DYNAMIC Thu Jun 27 18:11:45 UTC 2024"
}
```
- Cloud provider or hardware configuration:
- OS:
<!-- Eg., output of "cat /etc/os-release". -->
```
NAME="Fedora Linux"
VERSION="40.20240709.3.1 (CoreOS)"
ID=fedora
VERSION_ID=40
VERSION_CODENAME=""
PLATFORM_ID="platform:f40"
PRETTY_NAME="Fedora CoreOS 40.20240709.3.1"
ANSI_COLOR="0;38;2;60;110;180"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:40"
HOME_URL="https://getfedora.org/coreos/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora-coreos/"
SUPPORT_URL="https://github.com/coreos/fedora-coreos-tracker/"
BUG_REPORT_URL="https://github.com/coreos/fedora-coreos-tracker/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=40
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=40
SUPPORT_END=2025-05-13
VARIANT="CoreOS"
VARIANT_ID=coreos
OSTREE_VERSION='40.20240709.3.1'
```
- Kernel: `Linux REDACTED 6.9.7-200.fc40.x86_64 #1 SMP PREEMPT_DYNAMIC Thu Jun 27 18:11:45 UTC 2024 x86_64 GNU/Linux`
- Installation method: Kubernetes manifest

**Additional context**

<!-- Add any other context about the problem here. -->

--- a/pkg/driverbuilder/builder/fedora.go
+++ b/pkg/driverbuilder/builder/fedora.go
@@ -85,6 +85,13 @@ func (c *fedora) URLs(kr kernelrelease.KernelRelease) ([]string, error) {
 			kr.Fullversion,
 			kr.FullExtraversion,
 		),
+		fmt.Sprintf( // updates-archive
+			"https://fedoraproject-updates-archive.fedoraproject.org/fedora/%s/%s/kernel-devel-%s%s.rpm",
+			version,
+			kr.Architecture.ToNonDeb(),
+			kr.Fullversion,
+			kr.FullExtraversion,
+		),
 	}
 
 	// return out all possible urls

--- a/pkg/driverbuilder/local.go
+++ b/pkg/driverbuilder/local.go
@@ -103,6 +103,8 @@ func (lbp *LocalBuildProcessor) Start(b *builder.Build) error {
 				} else {
 					lbp.Logger.Warn("Failed to download headers.", lbp.Logger.Args("err", err))
 				}
+			} else {
+				lbp.Logger.Warn("Failed to generate script.", lbp.Logger.Args("err", err))
 			}
 		} else {
 			lbp.Logger.Info("Skipping kernel headers automatic download.", lbp.Logger.Args("err", err))

